From c12ea778654778f2039369323e8875e71d3a5609 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Mon, 29 Apr 2019 18:20:21 -0700
Subject: [PATCH] Linux 5.0 compat: Remove incorrect ASSERT

Not all block devices, notably scsi_debug, set a root_blkg on the
request queue.  Remove this assertion and allow the the existing
call to blkg_tryget() to gracefully handle the NULL (which it does).

Reviewed-by: Tomohiro Kusumi <kusumi.tomohiro@gmail.com>
Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
(backported from upstream commit c12ea778654778f2039369323e8875e71d3a5609)
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Closes #8678
---
 module/zfs/vdev_disk.c | 1 -
 1 file changed, 1 deletion(-)

Index: zfs-linux-0.7.12/module/zfs/vdev_disk.c
===================================================================
--- zfs-linux-0.7.12.orig/module/zfs/vdev_disk.c
+++ zfs-linux-0.7.12/module/zfs/vdev_disk.c
@@ -516,7 +516,6 @@ vdev_bio_associate_blkg(struct bio *bio)
 	struct request_queue *q = bio->bi_disk->queue;
 
 	ASSERT3P(q, !=, NULL);
-	ASSERT3P(q->root_blkg, !=, NULL);
 	ASSERT3P(bio->bi_blkg, ==, NULL);
 
 	if (blkg_tryget(q->root_blkg))
